proposed/tracing.md

```diff
@@ -10,14 +10,103 @@ interfaces and functionality first.
 
 [RFC 2119]: http://tools.ietf.org/html/rfc2119
 
-## Goal
+## 1. Definitions
 
-TBD
+* **Framework** - An application framework (or micro-framework) that runs a developers code.  
+  Eg. Laravel, Symphony, CakePHP, Slim
+* **Library** - Any library that developers may include that adds additional functionality.  
+  Eg. Image Manipulation, HTTP Clients, 3rd Party SDKs
+* **Provider** - An organization that offers APM (Application Performance Monitoring) as a service.
+  Typically, via a composer package
 
-## Definitions
+## 2. Interfaces
 
-* **Framework** - An application framework (or micro-framework) that runs a developers code.  
-  Eg. Laravel, Symfony, CakePHP, Slim
-* **Library** - Any library that developers may include that adds additional functionality  
-  Eg. Image Manipulation, HTTP Clients, 3rd Party SDK's 
-* **Provider** - An organization that offers APM as a service. Typically via a composer package
+## 2.1 Span
+
+See: [AAllport/psr-tracing - SpanInterface.php](https://github.com/AAllport/psr-tracing/blob/main/src/SpanInterface.php)
+
+A span represents a single operation within a trace. A trace is a collection of spans.
+A fully formed span will consist of the following data:
+
+- Name
+- Attributes, an associative array representing key-value pairs
+- Start Time
+- End Time
+- Outcome
+- Exceptions
+
+Upon creation, a new span will not be activated or have any attributes set.
+This allows the user to populate the span with data.
+It is RECOMMENDED that users supply a span with all relevant attributes shortly after creation, notably before
+activation.
+However, Providers SHOULD make every effort to persist attribute data if provided after a span is activated.
+
+To track the time a piece of work takes, a user will call `SpanInterface::activate()`.
+This will set the start time and push the current span into whichever context propagation system the provider chooses.
```

Comment:
Is this related to `getCurrentSpan()` from `TracerInterface`? After "activation" the tracer should return it?